src/message/MessageContainer.js

```diff
@@ -64,7 +64,7 @@ export default class MessageContainer extends React.PureComponent {
       pushRoute,
     } = this.props;
     const MessageComponent = isBrief ? MessageBrief : MessageFull;
-    const childrenNodes = htmlToDomTree(message.match_content || message.content);
+    const childrenNodes = htmlToDomTree((message.match_content || message.content).replace(/\r\n/g, '<br />').replace(/[\r\n]/g, '<br />'));
```

Comment:
There are two completely different sets of problems.

One is general style and markup simplification that can help the mobile app. That would be close to impossible to do in the mobile app, but easy to do server side. I am talking 100x the effort if done on the mobile app.

Some other changes like the ones we talk with @vishwesh3 here are more of a HTML+CSS spec related things. We don't need to change anything on the server for sure, as this is just too ad-hoc to expect the server to conform in some way to our expectations.

In this case we don't want to be replacing new lines with `<br>` tags as they behave differently outside `<pre>` tags.